enclave/src/contracts/balance.rs

```diff
@@ -0,0 +1,213 @@
+use std::collections::{BTreeMap};
+use serde::{de::{self,Visitor}, Serialize, Deserialize, Serializer, Deserializer};
+use core::str::FromStr;
+use core::cmp::Ord;
+use std::prelude::v1::*;
+use std::{fmt,vec::Vec};
+use std::str;
+
+use crate::contracts;
+use crate::types::TxRef;
+
+extern crate runtime as chain;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Balance{
+    accounts: BTreeMap<AccountIdWrapper, chain::Balance>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Command{
+    Transfer(TransferDetails),
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TransferDetails{
+    accounts: Vec<(AccountIdWrapper,chain::Balance)>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Request{
```

Comment:
Let's turn `Request` to an enum because we may have different types of requests.

```rust
pub enum Request {
    FreeBalance(FreeBalanceReq)
    // in the future: TotalIssuance,
}

pub struct FreeBalanceReq {
    account: AccountIdWrapper
}
```